caf-kubernetes-client/src/main/java/com/github/cafapi/kubernetes/client/KubernetesClientFactory.java

```diff
@@ -0,0 +1,165 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.cafapi.kubernetes.client;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.net.URI;
+import java.security.KeyStore;
+import java.security.SecureRandom;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateFactory;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManagerFactory;
+
+import org.glassfish.jersey.client.ClientConfig;
+import org.glassfish.jersey.jackson.internal.jackson.jaxrs.json.JacksonJsonProvider;
+import org.glassfish.jersey.jdk.connector.JdkConnectorProvider;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import com.github.cafapi.kubernetes.client.client.ApiClient;
+
+import jakarta.ws.rs.client.Client;
+import jakarta.ws.rs.client.ClientBuilder;
+
+/**
+ * This class is loosely based on
+ * <a href="https://github.com/kubernetes-client/java/blob/automated-release-21.0.1/util/src/main/java/io/kubernetes/client/util/ClientBuilder.java#L70">io.kubernetes.client.util.ClientBuilder.java</a>,
+ * which we stopped using because it brought in a lot of unwanted dependencies (such as the AWS SDK, Google Protocol Buffers, and the
+ * Bouncy Castle libraries).
+ */
+public final class KubernetesClientFactory
+{
+    private static final String SERVICEACCOUNT_ROOT = "/var/run/secrets/kubernetes.io/serviceaccount";
+    private static final String SERVICEACCOUNT_CA_PATH = SERVICEACCOUNT_ROOT + "/ca.crt";
+    private static final String SERVICEACCOUNT_TOKEN_PATH = SERVICEACCOUNT_ROOT + "/token";
+    private static final String ENV_SERVICE_HOST = "KUBERNETES_SERVICE_HOST";
+    private static final String ENV_SERVICE_PORT = "KUBERNETES_SERVICE_PORT";
+
+    private KubernetesClientFactory()
+    {
+    }
+
+    /**
+     * Creates a Kubernetes client with a CA cert and Bearer token.
+     * <p>
+     * This method expects the following environment variables to be present:
+     * <ul>
+     *   <li>KUBERNETES_SERVICE_HOST</li>
+     *   <li>KUBERNETES_SERVICE_PORT</li>
+     * </ul>
+     * <p>
+     * and the following files to be present:
+     * <ul>
+     *   <li>/var/run/secrets/kubernetes.io/serviceaccount/ca.crt</li>
+     *   <li>/var/run/secrets/kubernetes.io/serviceaccount/token</li>
+     * </ul>
+     *
+     * @return a client configured to communicate with Kubernetes using a CA cert and Bearer token.
+     * @throws Exception if the client could not be created for any reason.
+     */
+    public static ApiClient createClientWithCertAndToken() throws Exception
```

Comment:
I don't like that we're returning a general `Exception` here.  Let's add our own `FailedToCreateKubernetesClientException` for this to throw, and have it wrap the various checked exceptions.